### PR TITLE
[FRONTEND] Trending Products Review

### PR DESCRIPTION
**Pull request by furkan-varol**
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other

# Description

Product information added to Trending Products Component in the Home Page.

# Screenshots

image
image

# Test results

Tested with products coming from backend and works as intended.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have checked my code and corrected any misspellings
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: ecommerce/frontend/src/components/TrendingGrid.js [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
                     <Link to={'/product/' + product.id}>
-                        <img src={product.images[0]} className="gallery__img" alt="" />
+                        <img src={product.imageUrl} className="gallery__img" alt="" />
+                        <div class="product-info">
+                            <div class="product-name">{product.title}</div>
+                            <div class="product-price">
+                                {product.price}&nbsp;{product.currency}
+                            </div>
+                            <Rate disabled allowHalf defaultValue={product.rating}></Rate>
+                        </div>
```
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/TrendingGrid.js]: we use images, no longer imageUrl

File: ecommerce/frontend/src/components/TrendingGrid.js [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
             {trendingProducts.map((product, index) => (
                 <figure className={'gallery__item gallery__item--' + index} key={product.id}>
                     <Link to={'/product/' + product.id}>
-                        <img src={product.images[0]} className="gallery__img" alt="" />
+                        <img src={product.imageUrl} className="gallery__img" alt="" />
+                        <div class="product-info">
+                            <div class="product-name">{product.title}</div>
+                            <div class="product-price">
+                                {product.price}&nbsp;{product.currency}
+                            </div>
```
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/TrendingGrid.js]: products from backend do not have currency for now, needs backup

File: ecommerce/frontend/src/components/TrendingGrid.less [hunk 3]
```diff
@@ -6,21 +6,57 @@
...
+.gallery__item:hover .product-info {
+    opacity: 1;
+}
+
 .gallery__img {
+    opacity: 1;
+    display: block;
     width: 100%;
+    transition: .5s ease;
+    backface-visibility: hidden;
```
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/TrendingGrid.less]: what is backface visibility for? seems like for 3d stuff..

File: ecommerce/frontend/src/config.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 export const config = {
-    apiUrl: process.env.REACT_APP_API_URL ?? 'http://localhost:8000',
+    apiUrl: 'http://ec2-18-189-28-20.us-east-2.compute.amazonaws.com:8000', // process.env.REACT_APP_API_URL ?? 'http://localhost:8000',
 }
```
Comment (MehdiSaffar) [inline ecommerce/frontend/src/config.js]: do not change this otherwise won't work for different backends.

**Review by MehdiSaffar — CHANGES_REQUESTED** [review]

File: ecommerce/frontend/src/components/TrendingGrid.js (hunk 1 above)
Comment (furkan-varol) [reply ecommerce/frontend/src/components/TrendingGrid.js]: Changed to images[0].

File: ecommerce/frontend/src/components/TrendingGrid.js (hunk 2 above)
Comment (furkan-varol) [reply ecommerce/frontend/src/components/TrendingGrid.js]: formatProduct is used here. Issue fixed.

File: ecommerce/frontend/src/components/TrendingGrid.less (hunk 3 above)
Comment (furkan-varol) [reply ecommerce/frontend/src/components/TrendingGrid.less]: Removed.

File: ecommerce/frontend/src/config.js (hunk 4 above)
Comment (furkan-varol) [reply ecommerce/frontend/src/config.js]: Reverted back to original.

**Comment by furkan-varol** [comment]
Decided to abandon the component.

**Pull request closed without merging**